cirq/ops/common_channels.py

```diff
@@ -676,20 +636,13 @@ def __init__(self, p) -> None:
         Raises:
             ValueError: if p is not a valid probability.
         """
-
-        def validate_probability(p, p_str):
-            if p < 0:
-                raise ValueError('{} was less than 0.'.format(p_str))
-            elif p > 1:
-                raise ValueError('{} was greater than 1.'.format(p_str))
-            return p
-
-        self._p = validate_probability(p, 'p')
+        self._p = value.validate_probability(p, 'p')
         self._delegate = AsymmetricDepolarizingChannel(1. - p, 0., 0.)
 
-    def _channel_(self) -> Iterable[np.ndarray]:
-        # Return just the I and X pieces.
-        return list(self._delegate._channel_())[:2]
+    def _mixture_(self) -> Sequence[Tuple[float, np.ndarray]]:
+        mixture = protocols.mixture(self._delegate)
+        # just return identity and x term
+        return (mixture[0], mixture[1])
```

Comment:
This is a sneaky kind of way to couple this class to the implementation of `protocols.mixture`. Are we guaranteeing that it will never e.g. drop 0 terms?

Reply:
Moved over to just calling _mixture_ as it seems fine in this case.  I don't think we will drop 0 terms.